python_ta/checkers/redundant_or_impossible_condition_checker.py

```diff
@@ -0,0 +1,117 @@
+"""
+Check for redundant If or While conditions in functions based on z3 constraints
+"""
+
+from typing import Any, Union
+
+try:
+    import z3
+
+    z3_dependency_available = True
+except ImportError:
+    z3 = Any
+    z3_dependency_available = False
+
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+
+class RedundantOrImpossibleConditionChecker(BaseChecker):
+    name = "redundant-condition"
+    msgs = {
+        "R9900": (
+            """This conditional statement is already true given the preceding code;
```

Comment:
Reword to "This condition will always evaluate to True." Similarly for the `impossible-condition` error.

Note: As one of your examples above demonstrates, this checker will flag conditions that are tautological independent of the surrounding code, e.g. `x or not x`. So it's misleading to reference "preceding code" or, below, "preceding Z3 constraints".